server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -201,11 +201,9 @@ private TransferFileSnapshot prepareMetadata(TransferSnapshot transferSnapshot)
      *
      * @param primaryTerm primary term where the generations will be deleted.
      * @param generations set of generation to delete.
+     * @param onDeletion runnable to run on deletion of translog and metadata files
      */
-    public void deleteGenerationAsync(long primaryTerm, Set<Long> generations) {
-        if (generations.isEmpty()) {
-            return;
-        }
+    public void deleteGenerationAsync(long primaryTerm, Set<Long> generations, Runnable onDeletion) {
```

Comment:
we should rename `onDeletion` to `onCompletion` as we are doing it irrespective of success.